bundles/org.openhab.binding.avmfritz/src/main/java/org/openhab/binding/avmfritz/internal/callmonitor/CallMonitor.java

```diff
@@ -0,0 +1,202 @@
+            }
+
+            // create a new thread for listening to the FritzBox
+            logger.debug("Reconnecting to Fritzbox call monitor port.");
+            monitorThread = new CallMonitorThread();
+            monitorThread.start();
+        }, 0, 2, TimeUnit.HOURS);
+    }
+
+    /**
+     * Cancel the reconnect job.
+     */
+    public void dispose() {
+        reconnectJob.cancel(true);
+    }
+
+    public class CallMonitorThread extends Thread {
+
+        // Socket to connect
+        private @Nullable Socket socket;
+
+        // Thread control flag
+        private boolean interrupted = false;
+
+        // time to wait before reconnecting
+        private long reconnectTime = 60000L;
+
+        public CallMonitorThread() {
+        }
+
+        @Override
+        public void run() {
+            while (!interrupted) {
+                BufferedReader reader = null;
+                try {
+                    logger.debug("Callmonitor thread [{}] attempting connection to FritzBox on {}:{}.",
+                            Thread.currentThread().getId(), ip, MONITOR_PORT);
+                    socket = new Socket(ip, MONITOR_PORT);
+                    reader = new BufferedReader(new InputStreamReader(socket.getInputStream()));
+                    // reset the retry interval
+                    reconnectTime = 60000L;
+                } catch (Exception e) {
+                    logger.warn("Error attempting to connect to FritzBox. Retrying in {} seconds",
+                            reconnectTime / 1000L, e);
+                    try {
+                        Thread.sleep(reconnectTime);
+                    } catch (InterruptedException ex) {
+                        interrupted = true;
+                    }
+                    // wait another more minute the next time
+                    reconnectTime += 60000L;
+                }
+                if (reader != null) {
+                    logger.info("Connected to FritzBox call monitor at {}:{}.", ip, MONITOR_PORT);
+                    while (!interrupted) {
+                        try {
+                            String line = reader.readLine();
+                            if (line != null) {
+                                logger.debug("Received raw call string from fbox: {}", line);
+                                CallEvent ce = new CallEvent(line);
+                                handleCallEvent(ce);
+                            }
+                        } catch (IOException e) {
+                            if (interrupted) {
+                                logger.debug("Lost connection to Fritzbox because of an interrupt.");
+                            } else {
+                                logger.warn("Lost connection to FritzBox: {}", e.getMessage());
+                            }
+                            break;
+                        } finally {
+                            try {
+                                sleep(5000L);
```

Comment:
I think 5 seconds is really long reaction time for binding, Maybe this should be a configuration Parameter

Reply:
I changed it to 1 second. Don't think there's any need to make it configurable.